meeteval/io/base.py

```diff
@@ -20,12 +23,33 @@
     Subclasses = 'UEM | STM | CTM | RTTM'
 
 
+
+class BaseABC:
+    @classmethod
+    def new(cls, d, **defaults):
+        # Example code:
+        # from meeteval.io.seglst import asseglst
+        # seglst = asseglst(d).map(lambda s: {**s, **defaults})
+        # ... (conert seglst to cls)
+        raise NotImplementedError()
+
+    def to_seglst(self):
+        raise NotImplementedError()
+
+
 @dataclass(frozen=True)
 class BaseLine:
     @classmethod
     def parse(cls, line: str) -> 'Self':
         raise NotImplementedError(cls)
 
+    @classmethod
+    def from_seglst(cls, segment: 'SegLstSegment') -> 'Self':
```

Comment:
Change the name to `from_dict`? The input is an entry from a seglst, which is a dict with special keys.

Reply:
Done